lib/zinnia.js

```diff
@@ -251,6 +260,21 @@ export async function run ({
         }
       }
     })(),
+    (async () => {
+      while (true) {
+        const delay = 10 * 60 * 1000 // 10 minutes
+        const jitter = Math.random() * 20_000 - 10_000 // +- 10 seconds
+        try {
+          await timers.setTimeout(delay + jitter, null, { signal })
+        } catch (err) {
+          if (err.name === 'AbortError') return
+          throw err
+        }
+        const newContracts = await getContractsWithRetry(provider, abi)
+        while (contracts.length > 0) contracts.pop()
+        for (const contract of newContracts) contracts.push(contract)
```

Comment:
Nitpick: this can be simplified.

```suggestion
        contracts.splice(0)
        contracts.push(...newContracts)
```

Reply:
Or even like this:

```suggestion
        contracts.splice(0, Infinity, ...newContracts)
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
Great suggestion! I committed the first one after seeing it, I think the next one is cool but not worth the extra effort now